packages/desktop-client/src/components/reports/ReportOptions.tsx

```diff
@@ -44,3 +51,132 @@ const intervalOptions = [
 { value: 5, description: 'Yearly', name: 5,
 ];
 */
+export type QueryDataEntity = {
+  date: string;
+  category: string;
+  categoryGroup: string;
+  account: string;
+  accountOffBudget: boolean;
+  payee: string;
+  transferAccount: string;
+  amount: number;
+};
+
+export type UncategorizedEntity = {
+  id?: string;
+  name: string;
+  hidden?: boolean;
+  uncat_id: string;
+  offBudget: boolean;
+  transfer: boolean;
+  category: boolean;
+};
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏
```suggestion
export type UncategorizedEntity = CategoryEntity & {
  id: null;
  uncat_id: string;
  offBudget: boolean; // what does this do? naming could be improved; also: we have a mixture of `camelCase` and `snake_case` here - ideally should use just one
  transfer: boolean; // what does this do? naming could be improved
  category: boolean; // what does this do? naming could be improved
};
```